electrum/plugins/ledger/ledger.py

```diff
@@ -540,75 +764,548 @@ def sign_transaction(self, tx, password):
+        if client is None:
+            client = self.get_client()
+        return client
+
+    def decrypt_message(self, pubkey, message, password):
+        raise UserFacingException(_('Encryption and decryption are currently not supported for {}').format(self.device))
+
+    def sign_message(self, sequence, *args, **kwargs):
+        address_path = self.get_derivation_prefix()[2:] + "/%d/%d" % sequence
+        return self.get_client_dongle_object().sign_message(address_path, *args, **kwargs)
+
+    def sign_transaction(self, *args, **kwargs):
+        return self.get_client_dongle_object().sign_transaction(self, *args, **kwargs)
+
+    def show_address(self, sequence, *args, **kwargs):
+        address_path = self.get_derivation_prefix()[2:] + "/%d/%d" % sequence
+        return self.get_client_dongle_object().show_address(address_path, *args, **kwargs)
+
+
 class LedgerPlugin(HW_PluginBase):
     keystore_class = Ledger_KeyStore
-    minimum_library = (0, 1, 32)
-    DEVICE_IDS = [
-                   (0x2581, 0x1807), # HW.1 legacy btchip
-                   (0x2581, 0x2b7c), # HW.1 transitional production
-                   (0x2581, 0x3b7c), # HW.1 ledger production
-                   (0x2581, 0x4b7c), # HW.1 ledger test
-                   (0x2c97, 0x0000), # Blue
-                   (0x2c97, 0x0001), # Nano-S
-                   (0x2c97, 0x0004), # Nano-X
-                   (0x2c97, 0x0005), # Nano-S Plus
-                   (0x2c97, 0x0006), # RFU
-                   (0x2c97, 0x0007), # RFU
-                   (0x2c97, 0x0008), # RFU
-                   (0x2c97, 0x0009), # RFU
-                   (0x2c97, 0x000a)  # RFU
-                 ]
+    minimum_library = (0, 1, 1)
+    DEVICE_IDS = [(0x2581, 0x1807),  # HW.1 legacy btchip
+                  (0x2581, 0x2b7c),  # HW.1 transitional production
+                  (0x2581, 0x3b7c),  # HW.1 ledger production
+                  (0x2581, 0x4b7c),  # HW.1 ledger test
+                  (0x2c97, 0x0000),  # Blue
+                  (0x2c97, 0x0001),  # Nano-S
+                  (0x2c97, 0x0004),  # Nano-X
+                  (0x2c97, 0x0005),  # Nano-S Plus
+                  (0x2c97, 0x0006),  # RFU
+                  (0x2c97, 0x0007),  # RFU
+                  (0x2c97, 0x0008),  # RFU
+                  (0x2c97, 0x0009),  # RFU
+                  (0x2c97, 0x000a)]  # RFU
     VENDOR_IDS = (0x2c97,)
     LEDGER_MODEL_IDS = {
         0x10: "Ledger Nano S",
         0x40: "Ledger Nano X",
         0x50: "Ledger Nano S Plus",
     }
+
     SUPPORTED_XTYPES = ('standard', 'p2wpkh-p2sh', 'p2wpkh', 'p2wsh-p2sh', 'p2wsh')
 
     def __init__(self, parent, config, name):
         self.segwit = config.get("segwit")
         HW_PluginBase.__init__(self, parent, config, name)
         self.libraries_available = self.check_libraries_available()
         if not self.libraries_available:
+            print("Library unavailable")
```

Comment:
Not sure if there are cases where the `return` is still useful, so I replaced the `print` with the logger in 7c80779903008652103cc90885088ac85963a98a rather than removing it altogether.